platforms/software/publish/src/main/java/org/gradle/api/publish/internal/component/ConfigurationVariantMapping.java

```diff
@@ -108,12 +107,13 @@ private void visitVariant(
     }
 
     // Cannot be private due to reflective instantiation
-    static class DefaultConfigurationVariant implements ConfigurationVariant {
+    static abstract class DefaultConfigurationVariant implements ConfigurationVariant {
         private final ConfigurationInternal outgoingConfiguration;
 
         @Inject
         public DefaultConfigurationVariant(ConfigurationInternal outgoingConfiguration) {
             this.outgoingConfiguration = outgoingConfiguration;
+            getDescription().convention(outgoingConfiguration.getDescription());
```

Comment:
This type is meant to be immutable. 

Can we finalize getDescription() in the constructor here?